userspace/libsinsp/threadinfo.cpp

```diff
@@ -2023,7 +2027,15 @@ threadinfo_map_t::ptr_t sinsp_thread_manager::get_thread_ref(int64_t tid, bool q
         }
 
         scap_threadinfo scap_proc {};
-	bool have_scap_proc = false;
+        bool have_scap_proc = false;
+
+        // leaving scap_proc uninitialized could lead to undefined behaviour.
+        // to be safe we should initialized to zero.
+        memset(&scap_proc, 0, sizeof(scap_threadinfo));
+
+        scap_proc.tid = -1;
```

Comment:
Why we don't do this initialization to `-1` every time?

Reply:
This is just to be extra safe: in particular if we skip, for whatever reason, a part of the initialization, the threadinfo will be considered invalid.